scalyr_agent/log_processing.py

```diff
@@ -1399,6 +1399,12 @@ def __init__(self, file_path, config, log_config, close_when_staleness_exceeds=N
         # Tracks whether the processor has recently logged data
         self.__is_active = False
 
+        # Tracks whether or not the processor has been marked as finished
+        self.__is_finished = False
+
+        # Tracks the number of bytes that was last read from the processor
+        self.__prev_bytes_read = 0
```

Comment:
I think we can remove this if we move the check for when to close a processor based on the `__close_when_caught_up` to the completion callback in the `perform_processing` method.

Reply:
Done.  Though I changed `__close_when_caught_up`  to `__close_at_eof`